massa-execution-worker/src/interface_impl.rs

```diff
@@ -282,6 +305,8 @@ impl Interface for InterfaceImpl {
             operation_datastore: None,
         });
 
+        context.recursion_counter += 1;
```

Comment:
this creates an intrication that will increase future maintenance burden: developers will always need to keep in mind "should I also call `increment_recursion_counter` also when I call `init_call` from the runtime or not ?

Let's think of possible solutions

Reply:
@damip Now, we only increment/decrement the recursion counter in the sc-runtime, before and after calling `run_function`. This is clearer I think.